src/components/UserSelector.tsx

```diff
@@ -1,16 +1,69 @@
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import { User } from '../types/User';
+import cn from 'classnames';
+
+type Props = {
+  users: User[];
+  currUser: User | null;
+  setCurrUser: (currUser: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  currUser,
+  setCurrUser,
+}) => {
+  const [menuIsVisible, setMenuIsVisible] = useState(false);
+
+  const dropdownRef = useRef(null);
+
+  const onOutsideClick = (event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !(dropdownRef.current as HTMLElement).contains(event.target as Node)
+    ) {
+      setMenuIsVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', onOutsideClick);
+
+    return () => {
+      document.removeEventListener('mousedown', onOutsideClick);
+    };
+  }, []);
+
+  const selectUserFunc = (
+    e: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    user: User,
+  ) => {
+    e.preventDefault();
+    setCurrUser(user);
+    setMenuIsVisible(false);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div
+      data-cy="UserSelector"
+      ref={dropdownRef}
+      className={cn('dropdown', {
+        'is-active': users.length > 0 && menuIsVisible,
+      })}
+    >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setMenuIsVisible(true)}
```

Comment:
Consider toggling the `menuIsVisible` state instead of setting it to `true` directly. This will allow the button to both open and close the dropdown, enhancing user experience.